pkg/validate/security_context_linux.go

```diff
@@ -237,8 +244,9 @@ var _ = framework.KubeDescribe("Security Context", func() {
 
 			By("get nginx container pid")
 			command := []string{"cat", "/proc/1/cmdline"}
-			o := execSyncContainer(rc, containerID, command)
-			Expect(o).To(ContainSubstring("master process"))
+			Eventually(func() string {
+				return execSyncContainer(rc, containerID, command)
```

Comment:
Sometimes, especially on slow filesystems, `cat` is executed before nginx process change the contents of `/proc/1/cmdline` to the string containing `master process`. So we get no error from the execution itself but `cat` reports a string which doesn't meet the condition `Expect(o).To(ContainSubstring("master process"))`. So, same as the above, `Eventually` here is for automatic retrying rather than waiting. Please tell me if I'm missing something about the usage of `Eventually`.

> if there is a valid reason to loop should we not be looping inside execSyncContainer() vs around it?

Should we directly use `internalapi.RuntimeService.ExecSync` here?

Reply:
And thanks for your comments!

Reply:
I see now, thx. Did not realize what you meant by timing sensitive file creation. Did not think that the proc / pid info would not be populated at start of the container before the exec sync occurred.

If possible maybe you will want to consider looping in the container waiting on contents up to the 30sec timeout.. rather than exec multiple times. Either way this LGTM.

Reply:
Thanks for the comment. I fixed it to wait on the contents inside the container up to the timeout.